moveit_ros/hybrid_planning/test/test_basic_integration.cpp

```diff
@@ -0,0 +1,272 @@
+    global_solution_subscriber_ = node_->create_subscription<moveit_msgs::msg::MotionPlanResponse>(
+        "global_trajectory", rclcpp::SystemDefaultsQoS(),
+        [this](const moveit_msgs::msg::MotionPlanResponse::SharedPtr msg) {});
+
+    RCLCPP_INFO(node_->get_logger(), "Initialize Planning Scene Monitor");
+    tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node_->get_clock());
+
+    planning_scene_monitor_ = std::make_shared<planning_scene_monitor::PlanningSceneMonitor>(
+        node_, "robot_description", tf_buffer_, "planning_scene_monitor");
+    if (!planning_scene_monitor_->getPlanningScene())
+    {
+      RCLCPP_ERROR(node_->get_logger(), "The planning scene was not retrieved!");
+      std::exit(EXIT_FAILURE);
+    }
+    else
+    {
+      planning_scene_monitor_->startStateMonitor();
+      planning_scene_monitor_->providePlanningSceneService();  // let RViz display query PlanningScene
+      planning_scene_monitor_->setPlanningScenePublishingFrequency(100);
+      planning_scene_monitor_->startPublishingPlanningScene(planning_scene_monitor::PlanningSceneMonitor::UPDATE_SCENE,
+                                                            "/planning_scene");
+      planning_scene_monitor_->startSceneMonitor();
+    }
+
+    if (!planning_scene_monitor_->waitForCurrentRobotState(node_->now(), 5))
+    {
+      RCLCPP_ERROR(node_->get_logger(), "Timeout when waiting for /joint_states updates. Is the robot running?");
+      std::exit(EXIT_FAILURE);
+    }
+
+    if (!hp_action_client_->wait_for_action_server(20s))
+    {
+      RCLCPP_ERROR(node_->get_logger(), "Hybrid planning action server not available after waiting");
+      std::exit(EXIT_FAILURE);
+    }
+
+    // Setup motion planning goal taken from motion_planning_api tutorial
+    const std::string planning_group = "panda_arm";
+    robot_model_loader::RobotModelLoader robot_model_loader(node_, "robot_description");
+    const moveit::core::RobotModelPtr& robot_model = robot_model_loader.getModel();
+
+    // Create a RobotState and JointModelGroup
+    const auto robot_state = std::make_shared<moveit::core::RobotState>(robot_model);
+    const moveit::core::JointModelGroup* joint_model_group = robot_state->getJointModelGroup(planning_group);
+
+    // Configure a valid robot state
+    robot_state->setToDefaultValues(joint_model_group, "ready");
+
+    // Create desired motion goal
+    moveit_msgs::msg::MotionPlanRequest goal_motion_request;
+
+    moveit::core::robotStateToRobotStateMsg(*robot_state, goal_motion_request.start_state);
+    goal_motion_request.group_name = planning_group;
+    goal_motion_request.num_planning_attempts = 10;
+    goal_motion_request.max_velocity_scaling_factor = 0.1;
+    goal_motion_request.max_acceleration_scaling_factor = 0.1;
+    goal_motion_request.allowed_planning_time = 2.0;
+    goal_motion_request.planner_id = "ompl";
+    goal_motion_request.pipeline_id = "ompl";
```

Comment:
use inline lambda initialization, there seems to be no reason this should be mutable